src/coreComponents/fieldSpecification/SourceFluxStatistics.cpp

```diff
@@ -0,0 +1,364 @@
+                                                 WrappedStats const & wrappedStats )
+{
+  if( getLogLevel() >= minLogLevel && logger::internal::rank == 0 )
+  {
+    GEOS_LOG_RANK( GEOS_FMT( "{} {} (of {}, in {}): Producing on {} elements",
+                             catalogName(), getName(), wrappedStats.getFluxName(), elementSetName,
+                             wrappedStats.stats().m_elementCount ) );
+
+    // we want to format differently if we have got multiple phases or not
+    string_view massUnit = units::getSymbol( m_solver->getMassUnit() );
+    if( wrappedStats.stats().m_producedMass.size() == 1 )
+    {
+      GEOS_LOG_RANK( GEOS_FMT( "{} {} (of {}, in {}): Produced mass = {} {}",
+                               catalogName(), getName(), wrappedStats.getFluxName(), elementSetName,
+                               wrappedStats.stats().m_producedMass[0], massUnit ) );
+      GEOS_LOG_RANK( GEOS_FMT( "{} {} (of {}, in {}): Production rate = {} {}/s",
+                               catalogName(), getName(), wrappedStats.getFluxName(), elementSetName,
+                               wrappedStats.stats().m_productionRate[0], massUnit ) );
+    }
+    else
+    {
+      GEOS_LOG_RANK( GEOS_FMT( "{} {} (of {}, in {}): Produced mass = {} {}",
+                               catalogName(), getName(), wrappedStats.getFluxName(), elementSetName,
+                               wrappedStats.stats().m_producedMass, massUnit ) );
+      GEOS_LOG_RANK( GEOS_FMT( "{} {} (of {}, in {}): Production rate = {} {}/s",
+                               catalogName(), getName(), wrappedStats.getFluxName(), elementSetName,
+                               wrappedStats.stats().m_productionRate, massUnit ) );
+    }
+  }
+}
+
+void SourceFluxStatsAggregator::writeStatsToCSV( string_view elementSetName, WrappedStats const & stats,
+                                                 bool writeHeader )
+{
+  if( m_writeCSV > 0 && MpiWrapper::commRank() == 0 )
+  {
+    string const fileName = GEOS_FMT( "{}/{}_{}_{}.csv",
+                                      m_outputDir,
+                                      stats.getAggregatorName(), stats.getFluxName(), elementSetName );
+    std::ofstream outputFile( fileName,
+                              writeHeader ? std::ios_base::out : std::ios_base::app );
+    if( writeHeader )
+    {
+      outputFile << GEOS_FMT( "Time [s],Element Count,Producted Mass [{0}],Production Rate [{0}/s]",
+                              units::getSymbol( m_solver->getMassUnit() ) ) << std::endl;
+    }
+    else
+    {
+      outputFile << GEOS_FMT( "{},{},{},{}",
+                              stats.getStatsPeriodStart(), stats.stats().m_elementCount,
+                              stats.stats().m_producedMass, stats.stats().m_productionRate ) << std::endl;
+    }
+    outputFile.close();
+  }
+}
+
+bool SourceFluxStatsAggregator::execute( real64 const GEOS_UNUSED_PARAM( time_n ),
```

Comment:
Hi, I'd be interested in decoupling the gathering/aggregation of the data from its output.
Here we impose `log` and `csv`, for example. There's no reason not to use the `CSV` for other types of data, and to impose `CSV` for the source flux data.

@MelReyCG , I had a quick chat with @arng40 about that.